### PR TITLE
[Models]: Add MoM

### PR DESCRIPTION
**Pull request by WKX933**
### New Modules
- **`fla/layers/`**
  - Added `mom.py` 
  - Introduced `mom_varlen.py` (for varlen)

### Model Architecture
- **`models/`**
  - Implemented `mom_gated_deltanet` (Gated DeltaNet with MoM integration)

## Summary by CodeRabbit

* **New Features**
  * Added a Mixture-of-Memories (MoM) model family: memory-organized gated attention layer, Mom transformer blocks, MomConfig, MomModel, and MomForCausalLM with HF auto-model integration and generation/caching support.
  * Included an auxiliary load-balancing loss to encourage balanced routing across memory units.

* **Tests**
  * Added tests covering forward/backward passes and generation for the Mom model.

* **Chores**
  * Updated pytest path configuration for test discovery.

### CODE REVIEW

**Comment by zhiyuan1i** [comment]
@WKX933  Thank you for your PR!!
Could you please add tests in `tests/test_model.py` and `tests/test_generation.py` ?
And since I have introduece `l2warp` to all models, it would be good if you add some lines in MoM models, https://github.com/fla-org/flash-linear-attention/commit/82bb6496c1256981749462c7e83db07524e50656

File: fla/layers/mom_varlen.py
Comment (yzhangcs) [inline fla/layers/mom_varlen.py:1]: @WKX933 I think it would better if merging the two file `mom.py` and `mom_varlen.py`

File: fla/models/mom/configuration_mom.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        self.num_memories = num_memories
+        self.topk = topk
+        self.capacity = capacity
+        self.use_layer_wise_balance = use_layer_wise_balance
+        self.aux_loss_scale = aux_loss_scale
+        self.shared_mem = shared_mem
+        self.single_kv_proj = single_kv_proj
+        self.mom_backend = mom_backend
+
+        if not self.mom_backend in ['GDN']:
```
Comment (yzhangcs) [inline fla/models/mom/configuration_mom.py]: @JusenD Could we change the `mom_backend` name to `model_type` registered, e.g., `GatedDelteNetConfig.model_type`?

File: fla/layers/__init__.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 from .mamba import Mamba
 from .mamba2 import Mamba2
 from .mesa_net import MesaNet
+from .mom_varlen import MomGatedDeltaNet
```
Comment (yzhangcs) [inline fla/layers/__init__.py]: @JusenD Let's rename this module to `MomAttention`

File: fla/models/mom/modeling_mom.py [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+    supports_gradient_checkpointing = True
+    _no_split_modules = ['MomBlock']
+
+    def __init__(self, *inputs, **kwargs):
+        super().__init__(*inputs, **kwargs)
+
+    def _init_weights(
+        self,
+        module: nn.Module,
+        rescale_prenorm_residual: bool = True,
```
Comment (yzhangcs) [inline fla/models/mom/modeling_mom.py]: We should disable `rescale_prenorm_residual` by default.
